### ISSUE TITLE
Tracking issue: Update cert-manager manifests

### ISSUE THREAD

**Issue by DavidSpek** [kubeflow/manifests#1838]
As requested in https://github.com/kubeflow/manifests/issues/1234#issuecomment-818690526 here is a tracking issue for updating the cert-manager manifests. The manifests for cert-manager are severely outdated, originating from October 2019 and being version 0.11. At the time of writing cert-manger 1.3 is the most current version. To avoid comparability problems and for maintainers to need to spend time and effort supporting this outdated software it would be best to update cert-manager to the latest version and use the upstream manifests directly. This way, if there are problems with cert-manager or its installation issues can be redirected to cert-manager directly rather than taking up the time of the manifests working group.

https://github.com/kubeflow/manifests/pull/1820 Implements the changes suggested in this issue. 

/cc @yanniszark @kubeflow/wg-manifests-leads

**Comment by yanniszark** [comment]
Thanks a lot for opening an issue to track @DavidSpek!

**Comment by DavidSpek** [comment]
@yanniszark As mentioned in the the release channel on Slack, "KFserving requires certmanager to be version 0.12.0 or above". This means cert-manager should probably be upgrading.

**Comment by gecube** [comment]
I checked and I was able to install cert-manager directly from official manifest:
```
$ kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.3.0/cert-manager.yaml
```
I made it right before the installation of kubeflow.
But I found that in kubeflow manifest there is ClusterIssuer for KF in cert-manager component.
So we need to
1. disable `cert-manager-crds`, `cert-manager-kube-system-resources` and `cert-manager` components in KfDef file
2. manually create [ClusterIssuer for Kubeflow](https://github.com/kubeflow/manifests/blob/master/common/cert-manager/cert-manager/overlays/self-signed/cluster-issuer.yaml)

**Comment by DavidSpek** [comment]
@gecube If you look at my PR that is exactly what I am doing and includes the needed Kubeflow issuer.

**Comment by gecube** [comment]
@DavidSpek thanks for you good work! I hope we will be able to use it and see it in upstream ASAP.

**Comment by DavidSpek** [comment]
@gecube If you want to use it you can execute `kustomize build  github.com/argoflow/argoflow/cert-manager | kubectl apply -f -`.

**Comment by DavidSpek** [comment]
Just wanted to mention that cert-manager 1.4.0 was just released. I will update my PR accordingly.

**Comment by yanniszark** [comment]
Closed by #1916
